org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/nexus/SolsticeScanMonitor.java

```diff
@@ -297,13 +303,17 @@ private void addLinkToExternalFile(final NXcollection uniqueKeysCollection,
 	 * @param position
 	 * @throws Exception
 	 */
-	private void writePosition(IPosition position) throws Exception {
+	private void writePosition(IPosition position) {
 		if (!malcolmScan) {
 			IScanSlice rslice = IScanRankService.getScanRankService().createScanSlice(position);
 			SliceND sliceND = new SliceND(uniqueKeys.getShape(), uniqueKeys.getMaxShape(), rslice.getStart(), rslice.getStop(), rslice.getStep());
 			final int uniqueKey = position.getStepIndex() + 1;
 			final Dataset newActualPosition = DatasetFactory.createFromObject(uniqueKey);
-			uniqueKeys.setSlice(null, newActualPosition, sliceND);
+			try {
+				uniqueKeys.setSlice(null, newActualPosition, sliceND);
+			} catch (DatasetException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)